### PR TITLE
Fixes to COM wrappers

### PR DESCRIPTION
**Pull request by kant2002**
- Create IUnknown with 1 as ref count
- Add error message when cannot QI to desired interface

### CODE REVIEW

File: src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/ComWrappers.CoreRT.cs [hunk 1]
```diff
@@ -506,9 +506,11 @@ internal static unsafe IntPtr ComInterfaceForObject(object instance, Guid target
             int resultCode = wrapper->QueryInterface(in targetIID, out comObjectInterface);
             if (resultCode != 0)
             {
```
Comment (jkotas) [inline src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/ComWrappers.CoreRT.cs:511]: This path leaks the `unknownPtr` reference.

File: src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/ComWrappers.CoreRT.cs [hunk 2]
```diff
@@ -506,9 +506,11 @@ internal static unsafe IntPtr ComInterfaceForObject(object instance, Guid target
...
-                throw new PlatformNotSupportedException(SR.PlatformNotSupported_ComInterop);
+                throw new PlatformNotSupportedException(SR.InvalidOperation_InterfaceNotSupported);
```
Comment (jkotas) [inline src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/ComWrappers.CoreRT.cs]: Is this the exception that CoreCLR throws in this situation? (I would expect it to be something else.)

Comment (kant2002) [reply src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/ComWrappers.CoreRT.cs]: Cannot find location where it happens in CoreCLR. Will try again when would be home in hour or so. Will look at marshalling code.

Comment (kant2002) [reply src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/ComWrappers.CoreRT.cs]: Is this proper location where look for object -> interface conversion.

https://github.com/dotnet/runtime/blob/aaaaeed8c8d2d01dccd1f7fc536ea27eee75f472/src/coreclr/vm/ilmarshalers.cpp#L1401

Maybe Marshal class will throw this exception too.
I account for case where ComWrappers does not provide correct interface vtbl

Comment (jkotas) [reply src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/ComWrappers.CoreRT.cs]: The easiest way to find out is to write a small test.

File: src/coreclr/nativeaot/System.Private.CoreLib/src/Resources/Strings.resx [hunk 3]
```diff
@@ -3160,4 +3160,7 @@
   <data name="InvalidOperation_ComInteropRequireComWrapperInstance" xml:space="preserve">
     <value>COM Interop requires ComWrapper instance registered for marshalling.</value>
   </data>
+  <data name="InvalidOperation_InterfaceNotSupported" xml:space="preserve">
```
Comment (jkotas) [inline src/coreclr/nativeaot/System.Private.CoreLib/src/Resources/Strings.resx]: Could you please revert all changes in this file?

Comment (kant2002) [reply src/coreclr/nativeaot/System.Private.CoreLib/src/Resources/Strings.resx]: @jkotas done

**Review by jkotas — APPROVED** [review]
Thank you!

**Pull request merged**